helpers/lagoonenvclean.sh

```diff
@@ -0,0 +1,182 @@
+}
+
+#Function to retrieve environments per project
+allenvironment_query() {
+	echo "Lagoon $1 project"
+
+	# Substitute variables and escape characters in the query
+	ENVQUERY=$(echo $QLENVSQUERY | sed "s/PROJECT/$1/g" | sed 's/"/\\"/g' | sed 's/\\n/\\\\n/g' | awk -F'\n' '{if(NR == 1) {printf $0} else {printf "\\n"$0}}')
+
+	# Run the query
+	RESULT=`curl -s -k -X POST -H 'Content-Type: application/json' -H "$AMZBEARERTKN" $ENDPOINT -d "{\"query\": \"$ENVQUERY\"}"`
+
+	# From the results, take only environemnts called with a "pr-" to identify Pull Requests
+	ENVS=`echo $RESULT | jq -r '.[].projectByName.environments[].name'|grep -E -i "^(pr-)"`
+
+	# Retrieve the GitURL
+	GITURL=`echo $RESULT | jq -r '.[].projectByName.gitUrl'`
+
+	# Retrieve GitHub project and owner by the git url
+	GITPROJECT=`echo ${GITURL%%.git}|cut -f 2 -d "/"`
+	GITOWNER=`echo $GITURL | cut -f2 -d ":"|cut -f1 -d "/"`
+	
+	# Retrieve Lagoon project ID and Openshift name of the project
+	PROJECTID=`echo $RESULT | jq -r '.[].projectByName.id'`
+	OCPROJECT=`echo $RESULT | jq -r '.[].projectByName.environments[].openshiftProjectName'`
+
+	# Create a global associative array (it simulates a multi-dimensional array) where to store some variables per project
+	declare -gA LAGOONENVS
+	LAGOONENVS[$1,id]="$PROJECTID"
+	LAGOONENVS[$1,gitproject]="$GITPROJECT"
+	LAGOONENVS[$1,ocproject]="$OCPROJECT"
+	LAGOONENVS[$1,envs]="$ENVS"
+}
+
+# Function to retrieve information about single project's environment
+environment_query() {
+	
+	# Escape environment name
+	ENVCLEAN=$(echo $1 | sed -e "s#/#\\\/#g")
+
+	# Substitute and escape characters in GraphQL query
+	ONEENVQUERY=$(echo $QLENVQUERY | sed "s/NAME/\"$ENVCLEAN\"/g" | sed "s/ID/$2/g" | sed 's/"/\\"/g' | sed 's/\\n/\\\\n/g' | awk -F'\n' '{if(NR == 1) {printf $0} else {printf "\\n"$0}}')
+
+	# Run the query and store results into a variable
+	ONERESULT=`curl -s -k -X POST -H 'Content-Type: application/json' -H "$AMZBEARERTKN" $ENDPOINT -d "{\"query\": \"$ONEENVQUERY\"}"`
+
+	# Set variable with the last update of an environment, in a YYYYMMDD format
+	UPDATEDATE=`echo "$ONERESULT" | jq -r '.[].environmentByName.updated' | cut -f1 -d " "| xargs date +%Y%m%d -d`
+
+	# Declare a "fake" multi-demensional array, where to store variable per project and per environment
+	declare -gA LAGOONENV
+	LAGOONENV[$i,$1]="$UPDATEDATE"
+	#echo "Project $i, env $1, last update ${LAGOONENV[$i,$1]}"
+}
+
+
+# Function to delete Lagoon project's environment
+lagoonenv_clean() {
+
+	# Substitue and escape characters in GraphQL query
+	ENVDELETE=$(echo $QLENVDELETE | sed "s/ENV/\"$1\"/g" | sed "s/PROJECT/\"$2\"/g" | sed "s/BOOLEAN/\"$3\"/g" | sed 's/"/\\"/g' | sed 's/\\n/\\\\n/g' | awk -F'\n' '{if(NR == 1) {printf $0} else {printf "\\n"$0}}')
+	if [ "$3" = "false" ]; then
+		echo "Dry-run env delete"
+		echo $ENVDELETE
+	else
+		echo "Deleting the $1 env of $2 project"
+		echo $ENVDELETE
+	fi
+}
+
+# Function to query GitHub PR's status
+ghpr_query() {
```

Comment:
this function does not only query the github API, but it actually runs `lagoonenv_clean`, therefore I would suggest to call the function something that actually suggest that the clean will happen in here